generate_brick.py

```diff
@@ -242,6 +242,53 @@ def add_definitions():
             if len(definition[2]):
                 G.add((term, RDFS.seeAlso, URIRef(definition[2])))
 
+    qstr = """
+    select ?param where {
+      ?param rdfs:subClassOf* brick:Parameter.
+    }
+    """
+    params = [row["param"] for row in G.query(qstr)]
+    for param in params:
+        words = param.split("#")[-1].split("_")
+        prefix = words[0]
+        suffix = words[-1]
+        limit_def_template = "A parameter that places {direction} bound on the range of permitted values of a {setpoint}."
+
+        if suffix == "Limit":
+            if prefix == "Min":
+                direction = "a lower"
+            elif prefix == "Max":
+                direction = "an upper"
+            else:
+                prefix = None
+                direction = "a lower or upper"
+
+            if param.split("#")[-1] in ["Max_Limit", "Min_Limit", "Limit"]:
+                setpoint = "Setpoint"
+            else:
+                if prefix:
+                    setpoint = "_".join(words[1:-1])
+                else:
+                    setpoint = "_".join(words[:-1])
+
+            if setpoint.split("_")[-1] != "Setpoint":
+                setpoint = setpoint + "_Setpoint"
+                print(f"Created setpoint: {setpoint}")
```

Comment:
Can you leave some comments clarifying what this is doing? I'm not sure why we are parsing out the word "Setpoint" from the class name. Also in general `print` statements should be replaced with calls to `logging`

Reply:
Fixed this in the current commit.

`Limits` are associated with Setpoints, but some class names do not explicitly have associated `Setpoints` in their names. Thus, I manually add them here as a first attempt to fix such issues. I am open to suggestions. E.g., I say the definition of `Close_Limit` is "a boundary of `Close_Setpoint`", and throws a warning if `Close_Setpoint` does not exist.

Reply:
Isn't this backwards? I thought the idea was to take the set of `Setpoint` classes that are defined and then generate the `Limit` classes for those --- rather than attempting to align the two sets of classes. This way, the set of `Limit` classes is always aligned to the `Setpoint` classes we have defined.

Reply:
I intended to automatically generate definitions for existing Limits. I think not all the Setpoints should be explicitly associated with Limits.

Reply:
Okay, sounds good. We'll need the autogeneration of definitions either way